src/dataflow/src/source/mod.rs

```diff
@@ -1328,6 +1303,17 @@ where
             &consistency_info,
         );
 
+        let mut timestamp_bindings = if let TimestampDataUpdate::BringYourOwn(history) =
+            timestamp_histories
+                .borrow_mut()
+                .get(&id.source_id)
+                .expect("known to exist")
+        {
+            history.clone()
+        } else {
```

Comment:
this is pretty janky. trying to do this piecemeal and only look at byo sources for now. wanted to do it with options but kept running into weird lifetime issues